src/cloudformation_cli_python_lib/resource.py

```diff
@@ -145,15 +197,23 @@ def _parse_request(
 
     @_ensure_serialize
     def __call__(
-        self, event_data: MutableMapping[str, Any], _context: Any
+        self, event_data: MutableMapping[str, Any], context: LambdaContext
     ) -> MutableMapping[str, Any]:
         try:
             ProviderLogHandler.setup(event_data)
             parsed = self._parse_request(event_data)
             session, request, action, callback_context = parsed
-            progress_event = self._invoke_handler(
-                session, request, action, callback_context
-            )
+            invoke = True
+            while invoke:
+                progress_event = self._invoke_handler(
+                    session, request, action, callback_context
+                )
+                invoke = self.schedule_reinvocation(
+                    HandlerRequest.deserialize(event_data),
```

Comment:
wouldn't it be easier to just adapt `_parse_request` to return this? it already parses it